lib/testsuite/hooks/_basehook.js

```diff
@@ -90,12 +90,9 @@ class BaseHook {
             }
 
             if (this.context.queue && this.context.queue.inProgress) {
-              process.nextTick(() => {
-              
-                this.context.queue.once('queue:finished', _ => {
-                  runnableDone && runnableDone(result);
-                  doneFn(result);
-                });
+              this.context.queue.once('queue:finished', _ => {
```

Comment:
the delay here was added as an work around for another issue, so we shouldn't remove it.

Reply:
maybe we can split them, as in call the `doneFn()` immediately and leave the `runnableDone` with the delay? I think the issue reported here would be fixed.

Reply:
I think it will still wait for the commands to be executed as it will call the done after the queue has been finished. The problem with keeping the event listener in `process.nextTick` is, In case `done` is supplied to a callback of the Nightwatch command the event listener is attached after the completion of all nodes.  So the event is emitted before the event-listener over here is attached.

Reply:
@swrdfish any update on this?

Reply:
@gravityvi I came across this case where the execution hangs, I have not debugged it thoroughly, maybe you can take a look?

_ecosia.js_
```js
describe('Ecosia.org Demo', function() {

  before(function() {
    browser
      .fake();
  });

  it('Demo test ecosia.org', function(browser) {
    browser
      .navigateTo('https://www.ecosia.org/')
      .waitForElementVisible('body')
      .assert.titleContains('Ecosia')
      .assert.visible('input[type=search]')
      .setValue('input[type=search]', 'nightwatch')
      .assert.visible('button[type=submit]')
      .click('button[type=submit]')
      .assert.textContains('.layout__content', 'Nightwatch.js');
  });

  after(browser => browser.end());
});
```

_fake.js_
```js
module.exports = {
  command: function () {
    console.log('== custom command ==');
  }
};
```

Reply:
This might be unrelated to your fix, because your fix seems to be logically correct.

Reply:
@swrdfish Yes, this is unrelated to this PR. This is a limitation of functional style custom commands. If you convert it to class style command it will work fine. This problem exists in the main branch too.

Reply:
cool